src/plugin.js

```diff
@@ -174,6 +174,12 @@ const contribAdsPlugin = function(options) {
 
     // TODO reset state to content-set here instead of in every contentupdate case
     reset() {
+      if (player.ads.snapshot.trackChangeHandler) {
```

Comment:
Sorry but I'm going to have to disagree with Lahiru here. Not only is this not maintaining encapsulation, but having the same cleanup happen in two places makes the code difficult to reason about.